# Profitable apps profile for app store and google play store

This notebook is an exploration of the result of data apps from app store and google play store.

 i use A data set containing data about approximately 10,000 Android apps from Google Playa and  7,000 iOS apps  data from the App Store

with the data set, i try to determine the kinds of apps that are likely to attract more users 

## Opening Data and exploring data

In [1]:
#first, let's make function to open the dataset

def open_data(data, header = True):
    #read_in_data
    opendata = open(data)
    from csv import reader
    read_data = reader(opendata)
    
    #transform data into list of lists
    new_data = list(read_data)
    if header:
        return new_data[0], new_data[1:]
    
    return new_data

In [2]:
#open apple dataset
applestore_data_header, applestore_data = open_data('AppleStore.csv')

#open google dataset
googleplay_data_header, googleplay_data = open_data('googleplaystore.csv')

we can print many rows dataset using function below

In [3]:
#let's make function for explore dataset
def explore_data(dataset, start, end, rows_and_columns=False):
    dataset_slice = dataset[start:end]
    
    #print row in dataset_slice
    for row in dataset_slice:
        print(row)
        print('\n')
        
    #print number of row if rows_and columns true
    if rows_and_columns:
        print('number of rows: ', len(dataset))
        print('number of columns: ', len(dataset[0]))

In [4]:
#print a slice of applestore dataset
print(applestore_data_header)
print('\n')

data_explore = explore_data(applestore_data, 0,5, True)
print(data_explore)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


['420009108', 'Temple Run', '65921024', 'USD', '0.0', '1724546', '3842', '4.5', '4.0', '1.6.2', '9+', 'Games', '40', '5', '1', '1']


['284035177', 'Pandora - Music & Radio', '130242560', 'USD', '0.0', '1126879', '3594', '4.0', '4.5', '8.4.1', '12+', 'Music', '37', '4', '1', '1']


number of rows:  7197
number of columns:  16
None


In [5]:
#print a slice of googleplay data
print(googleplay_data_header)
print('\n')

data_explore_2 = explore_data(googleplay_data, 0,5, True)
print(data_explore_2)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Eve

## cleaning the data

for this sub-notebook i will detect inaccurate data and correct or remove it/
Detect duplicate data, and remove the duplicates

### delete incorrect data

In [6]:
#print many row dataset if there was incorrect data
print(googleplay_data_header)
print('\n')
print('data incorrect')
print(googleplay_data[10472])
print('\n')
print('data correct')
print(googleplay_data[0])

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


data incorrect
['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']


data correct
['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


in data , id 10472, there's missing value in rating, so we should delete it

In [7]:
print(len(googleplay_data))

#using function delete
del(googleplay_data[10472])
print(len(googleplay_data))

10841
10840


### delete duplicate entries

if we anlyze the data clearly, we will find many duplicate data, for instance

In [8]:
for app in googleplay_data:
    name = app[0]
    if name == 'Instagram':
        print(app)

['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577446', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66509917', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


to solve data duplicate problem, first we should search to all data set if there are another duplicate data

In [9]:
#function to find duplicate data
def find_duplicate(data):
    duplicate_app =[]
    unique_app =[]
    
    #if name not in duplicate_app , input name into unique_app, if in name in duplicate data input name into dupicate app
    for app in data:
        name = app[0]
        if name in unique_app:
            duplicate_app.append(name)
        else:
            unique_app.append(name)
            
    return duplicate_app, unique_app
            

In [10]:
#find duplicate in googleplay
google_duplicate_data, google_unique_data = find_duplicate(googleplay_data)

#print len to now number of dupicate data
print('googlestore')
print('number of duplicate apps: ', len(google_duplicate_data))
print('examples: ', google_duplicate_data[:15])

googlestore
number of duplicate apps:  1181
examples:  ['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings', 'Box', 'Zenefits', 'Google Ads', 'Google My Business', 'Slack', 'FreshBooks Classic', 'Insightly CRM', 'QuickBooks Accounting: Invoicing & Expenses', 'HipChat - Chat Built for Teams', 'Xero Accounting Software']


In [11]:
#find duplicate in apple store
apple_duplicate_data, apple_unique_data = find_duplicate(applestore_data)


print('applestore')
print('number of duplicate apps: ', len(apple_duplicate_data))
print('examples: ', apple_duplicate_data[:15])

applestore
number of duplicate apps:  0
examples:  []


i can delete duplicate data randomly, but we probably find another better way,

if we see in the previous 'instagram duplicate data, we know that almost every have same value except number of ratings, this column was very depend on time ( have different time) , more higher number of ratings, more higher time to get that number, more higher number  also more realible.

so using that analyze, i can use number of ratings column , to decide which row will be delete, and for that, row with the highest number will be keep, and other will be delete

To remove the duplicates, we will:

Create a dictionary, where each dictionary key is a unique app name and the corresponding dictionary value is the highest number of reviews of that app.

Use the information stored in the dictionary and create a new data set, which will have only one entry per app (and for each app, we'll only select the entry with the highest number of reviews).

In [12]:
#create dictionary
reviews_max = {}

#loop the row in dataset
for app in googleplay_data:
    name = app[0]
    n_reviews = float(app[3])
    if name in reviews_max and reviews_max[name] < n_reviews:
        reviews_max[name] = n_reviews
    elif name not in reviews_max :
        reviews_max[name] = n_reviews
        
print(len(reviews_max))
        
        

9659


In [13]:
#clean data from duplicate data
googleplay_clean =[]
already_added = []

for apps in googleplay_data:
    name = apps[0]
    n_reviews = float(apps[3])
    
    if (reviews_max[name] == n_reviews) and (name not in already_added):
        googleplay_clean.append(apps)
        already_added.append(name)
    
print(len(googleplay_clean))

9659


In [14]:

explore_data(googleplay_clean, 0, 5, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Everyone', 'Art & Design;Creativity', 'June 20, 2018', '1.1', '4.4 and up']


['Paper flowers instructions', 'ART_AND_DESIGN', '4.4', '167', '5.6M', '50,000+', 'Free', '0', 'Everyone', 'Art & Design', 'March 26, 2017', '1.0', '2.3 and up']


number of rows:  9659
number of columns:  13


### Remove non english apps
Recall that at our company, we only build apps that are free to download and install, and that are directed toward an English-speaking audience. This means that we'll need to:

Remove non-English apps like 爱奇艺PPS -《欢乐颂2》电视剧热播.
Remove apps that aren't free.

In [15]:
#create function to detect which apps are not english
#an alphabet or character which use in english-speaking audieance have ord between 1-127 so if ord higher than 127 is should be not alphabet
def is_english(a_string):
    for char in a_string:
        if ord(char) > 127:
            return False
        
    return True
    
print(is_english('Instagram'))
print(is_english('爱奇艺PPS -《欢乐颂2》电视剧热播'))
print(is_english('Docs To Go™ Free Office Suite'))
print(is_english('Instachat 😜'))


True
False
False
False


To minimize the impact of data loss, we'll only remove an app if its name has more than three characters with corresponding numbers falling outside the ASCII range. 

In [16]:

def is_english(a_string):
    non_ascii = 0
    for char in a_string:
        if ord(char) > 127:
            non_ascii += 1
    
    if non_ascii > 3:
        return False
    else:
        return True
    
print(is_english('Instagram'))
print(is_english('爱奇艺PPS -《欢乐颂2》电视剧热播'))
print(is_english('Docs To Go™ Free Office Suite'))

english = is_english('Instachat 😜')
print(english)

    

True
False
True
True


In [17]:
#clean data from non_english data
googleplay_english = []
for app in googleplay_clean:
    name = app[0]
    name_english = is_english(name)
    if name_english == True:
        googleplay_english.append(app)
        
        
applestore_english = []
for apps in applestore_data:
    name = apps[1]
    name_english = is_english(name)
    if name_english == True:
        applestore_english.append(apps)

explore_data(googleplay_english, 0, 5, True)
print('/n')
explore_data(applestore_english, 0, 5, True)
        
    
        

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Everyone', 'Art & Design;Creativity', 'June 20, 2018', '1.1', '4.4 and up']


['Paper flowers instructions', 'ART_AND_DESIGN', '4.4', '167', '5.6M', '50,000+', 'Free', '0', 'Everyone', 'Art & Design', 'March 26, 2017', '1.0', '2.3 and up']


number of rows:  9614
number of columns:  13
/n
['284882215', 'Facebook', '389879808', 'USD

## Isolating Free apps

in this notebook i will focus on free apps

In [18]:
#make new list with price =0
googleplay_final =[]
for app in googleplay_english:
    price = app[7]
    if price == '0':
        googleplay_final.append(app)
        
applestore_final = []        
for apps in applestore_english:
    price = apps[4]
    if price == '0.0':
        applestore_final.append(apps)

print(len(googleplay_final))
print(len(applestore_final))
        

8864
3222


as we see it before, my aim to determine the kinds of apps that we're likely can attract people to download it, to know about that, we should explore data, trying too see what kind apps which is most of people want to download it using frequency, depends on genres in both of ios apps dataset and google apps dataset.

To minimize risks and overhead, our validation strategy for an app idea is comprised of three steps:

Build a minimal Android version of the app, and add it to Google Play.
If the app has a good response from users, we develop it further.
If the app is profitable after six months, we build an iOS version of the app and add it to the App Store.


# Most Common apps by Genre

 to know , what kind of apps that will be more attracted by people , i focus on prime_genre column in the app store dataset and genres, category columns of the googleplay dataset

we will build two function
1. one function to generate frequency tables that show percentages
2. other function we can use to display the percentages in a descending order

In [19]:
#make frequency table
def freq_table(dataset, index):
    freq_dictionary = {}
    total = 0
    for app in dataset:
        total += 1
        data = app[index]
        if data in freq_dictionary:
            freq_dictionary[data] += 1
        else:
            freq_dictionary[data] = 1
            
    table_percentages = {}
    for key in freq_dictionary:
        percentages = (freq_dictionary[key] / total) * 100
        percent_round = round(percentages, 4)
        table_percentages[key] = percent_round
    
    return table_percentages

In [20]:
#def display dataset
import pandas as pd
def display_table(dataset, index):
    table = freq_table(dataset, index)
    table_display =[]
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)
        
    #sorted the data from the highest
    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ': ', entry[0])

In [21]:
applestore_data_header

['id',
 'track_name',
 'size_bytes',
 'currency',
 'price',
 'rating_count_tot',
 'rating_count_ver',
 'user_rating',
 'user_rating_ver',
 'ver',
 'cont_rating',
 'prime_genre',
 'sup_devices.num',
 'ipadSc_urls.num',
 'lang.num',
 'vpp_lic']

In [22]:
#using display_table to know number of apps depends on prime_genre
display_table(applestore_final, -5)

Games :  58.1626
Entertainment :  7.8833
Photo & Video :  4.9659
Education :  3.6623
Social Networking :  3.2899
Shopping :  2.6071
Utilities :  2.514
Sports :  2.1415
Music :  2.0484
Health & Fitness :  2.0174
Productivity :  1.7381
Lifestyle :  1.5829
News :  1.3346
Travel :  1.2415
Finance :  1.1173
Weather :  0.869
Food & Drink :  0.807
Reference :  0.5587
Business :  0.5276
Book :  0.4345
Navigation :  0.1862
Medical :  0.1862
Catalogs :  0.1241


in applestore_dataset, depend on prime_genre column, the most common genre is genre games with 58 % and following by entertainment with 8 %.

based on the frequency table, most of the people  are likely download games apps rather than other genre for their choice, it can be seen that difference between games and the second genre almost 49 %, 

the general impression that, most of genre is about fun(games, entertainment, social networking) while apps for practical purpose are more rare. however, the fact that fun apps are the most numerous doesn't also imply that they also have the greatest number of users — the demand might not be the same as the offer. 

In [23]:
googleplay_data_header

['App',
 'Category',
 'Rating',
 'Reviews',
 'Size',
 'Installs',
 'Type',
 'Price',
 'Content Rating',
 'Genres',
 'Last Updated',
 'Current Ver',
 'Android Ver']

In [24]:
googleplay_final[0:1]

[['Photo Editor & Candy Camera & Grid & ScrapBook',
  'ART_AND_DESIGN',
  '4.1',
  '159',
  '19M',
  '10,000+',
  'Free',
  '0',
  'Everyone',
  'Art & Design',
  'January 7, 2018',
  '1.0.0',
  '4.0.3 and up']]

In [25]:
#let's analysis googleplay data
#depens on category
display_table(googleplay_final,1)

FAMILY :  18.9079
GAME :  9.7247
TOOLS :  8.4612
BUSINESS :  4.5916
LIFESTYLE :  3.9034
PRODUCTIVITY :  3.8921
FINANCE :  3.7004
MEDICAL :  3.5311
SPORTS :  3.3958
PERSONALIZATION :  3.3168
COMMUNICATION :  3.2378
HEALTH_AND_FITNESS :  3.0799
PHOTOGRAPHY :  2.9445
NEWS_AND_MAGAZINES :  2.7978
SOCIAL :  2.6625
TRAVEL_AND_LOCAL :  2.3353
SHOPPING :  2.245
BOOKS_AND_REFERENCE :  2.1435
DATING :  1.8615
VIDEO_PLAYERS :  1.7938
MAPS_AND_NAVIGATION :  1.3989
FOOD_AND_DRINK :  1.241
EDUCATION :  1.162
ENTERTAINMENT :  0.9589
LIBRARIES_AND_DEMO :  0.9364
AUTO_AND_VEHICLES :  0.9251
HOUSE_AND_HOME :  0.8236
WEATHER :  0.801
EVENTS :  0.7107
PARENTING :  0.6543
ART_AND_DESIGN :  0.6431
COMICS :  0.6205
BEAUTY :  0.5979


In [26]:
#depend on genre
display_table(googleplay_final,-4)

Tools :  8.4499
Entertainment :  6.0695
Education :  5.3475
Business :  4.5916
Productivity :  3.8921
Lifestyle :  3.8921
Finance :  3.7004
Medical :  3.5311
Sports :  3.4634
Personalization :  3.3168
Communication :  3.2378
Action :  3.1024
Health & Fitness :  3.0799
Photography :  2.9445
News & Magazines :  2.7978
Social :  2.6625
Travel & Local :  2.324
Shopping :  2.245
Books & Reference :  2.1435
Simulation :  2.042
Dating :  1.8615
Arcade :  1.8502
Video Players & Editors :  1.7712
Casual :  1.7599
Maps & Navigation :  1.3989
Food & Drink :  1.241
Puzzle :  1.1282
Racing :  0.9928
Role Playing :  0.9364
Libraries & Demo :  0.9364
Auto & Vehicles :  0.9251
Strategy :  0.9138
House & Home :  0.8236
Weather :  0.801
Events :  0.7107
Adventure :  0.6769
Comics :  0.6092
Beauty :  0.5979
Art & Design :  0.5979
Parenting :  0.4964
Card :  0.4513
Casino :  0.4287
Trivia :  0.4174
Educational;Education :  0.3949
Board :  0.3836
Educational :  0.3723
Education;Education :  0.3384
Word :  

from the category and genres column, we know that is different from apps store market. In google apps dataset, most of people are likely choose practical purpose apps than fun apps, it can be seen in the first 10 category, that, only games includes fun apps.

it also follow by  genres column, which is only entertainment in first 10 genres, who includes fun apps,

other different is, google apps dataset have more balanced landcape of both practical and fun apps, proven by the different value between fun apps and practical apps

One way to find out what genres are the most popular (have the most users) is to calculate the average number of installs for each app genre. For the Google Play data set, we can find this information in the Installs column, but this information is missing for the App Store data set. As a workaround, we'll take the total number of user ratings as a proxy, which we can find in the rating_count_tot app.

### popular apps in applestore

In [27]:
#make frequent table
applestore_freq_table = freq_table(applestore_final, -5)
dict_avg = {}

#loop genre
for genre in applestore_freq_table:
    total = 0
    len_genre = 0
    
    #loop app in applestore_data
    for app in applestore_final:
        genre_app = app[-5]
        
        #if genre = genre in appsstore, sum n_ratings and number rows with that genre
        if genre_app == genre:
            n_ratings = float(app[5])
            total += n_ratings
            len_genre += 1
    
    #create average
    avg_ratings = total / len_genre
    dict_avg[genre] = avg_ratings
    
    #make new list, this is have function to sorted the data from highest
    table_avg_tuple = []
    for key in dict_avg:
        key_val_as_tuple = (dict_avg[key], key)
        table_avg_tuple.append(key_val_as_tuple)
        
    #sorted table
    table_sorted = sorted(table_avg_tuple, reverse = True)
   
#print the data
for x in table_sorted:
    print(x[1], ':', x[0])
        

Navigation : 86090.33333333333
Reference : 74942.11111111111
Social Networking : 71548.34905660378
Music : 57326.530303030304
Weather : 52279.892857142855
Book : 39758.5
Food & Drink : 33333.92307692308
Finance : 31467.944444444445
Photo & Video : 28441.54375
Travel : 28243.8
Shopping : 26919.690476190477
Health & Fitness : 23298.015384615384
Sports : 23008.898550724636
Games : 22788.6696905016
News : 21248.023255813954
Productivity : 21028.410714285714
Utilities : 18684.456790123455
Lifestyle : 16485.764705882353
Entertainment : 14029.830708661417
Business : 7491.117647058823
Education : 7003.983050847458
Catalogs : 4004.0
Medical : 612.0


on average, navigation have the biggest number user rating,  to know more what kind of apps is this, we can analyze it


In [28]:
for app in applestore_final:
    if app[-5] == 'Navigation':
        print(app[1], ':', app[5])

Waze - GPS Navigation, Maps & Real-time Traffic : 345046
Google Maps - Navigation & Transit : 154911
Geocaching® : 12811
CoPilot GPS – Car Navigation & Offline Maps : 3582
ImmobilienScout24: Real Estate Search in Germany : 187
Railway Route Search : 5


for the second largest

In [29]:
for app in applestore_final:
    if app[-5] == 'Reference':
        print(app[1], ':', app[5])

Bible : 985920
Dictionary.com Dictionary & Thesaurus : 200047
Dictionary.com Dictionary & Thesaurus for iPad : 54175
Google Translate : 26786
Muslim Pro: Ramadan 2017 Prayer Times, Azan, Quran : 18418
New Furniture Mods - Pocket Wiki & Game Tools for Minecraft PC Edition : 17588
Merriam-Webster Dictionary : 16849
Night Sky : 12122
City Maps for Minecraft PE - The Best Maps for Minecraft Pocket Edition (MCPE) : 8535
LUCKY BLOCK MOD ™ for Minecraft PC Edition - The Best Pocket Wiki & Mods Installer Tools : 4693
GUNS MODS for Minecraft PC Edition - Mods Tools : 1497
Guides for Pokémon GO - Pokemon GO News and Cheats : 826
WWDC : 762
Horror Maps for Minecraft PE - Download The Scariest Maps for Minecraft Pocket Edition (MCPE) Free : 718
VPN Express : 14
Real Bike Traffic Rider Virtual Reality Glasses : 8
教えて!goo : 0
Jishokun-Japanese English Dictionary & Translator : 0


for the third largest


In [30]:
for app in applestore_final:
    if app[-5] == 'Social Networking':
        print( app[1], ':', app[5])

Facebook : 2974676
Pinterest : 1061624
Skype for iPhone : 373519
Messenger : 351466
Tumblr : 334293
WhatsApp Messenger : 287589
Kik : 260965
ooVoo – Free Video Call, Text and Voice : 177501
TextNow - Unlimited Text + Calls : 164963
Viber Messenger – Text & Call : 164249
Followers - Social Analytics For Instagram : 112778
MeetMe - Chat and Meet New People : 97072
We Heart It - Fashion, wallpapers, quotes, tattoos : 90414
InsTrack for Instagram - Analytics Plus More : 85535
Tango - Free Video Call, Voice and Chat : 75412
LinkedIn : 71856
Match™ - #1 Dating App. : 60659
Skype for iPad : 60163
POF - Best Dating App for Conversations : 52642
Timehop : 49510
Find My Family, Friends & iPhone - Life360 Locator : 43877
Whisper - Share, Express, Meet : 39819
Hangouts : 36404
LINE PLAY - Your Avatar World : 34677
WeChat : 34584
Badoo - Meet New People, Chat, Socialize. : 34428
Followers + for Instagram - Follower Analytics : 28633
GroupMe : 28260
Marco Polo Video Walkie Talkie : 27662
Miitomo : 2

if we focus on third largest, it's look like average not give  us real information, for exmple, eventhough, reference become second largest average, bible have number of rating greater than waze and google maps.

also, there are big skewed in reference and navigation, while networking more balanced landscape

if we see at that three genres, we will know that, most of apps in social networking and navigation, which are have biggest number of ratings belongs to big company. while reference not belongs to that company.

because the first and the third dominated by apps from big company, it's look like not potential to make apps with have same genre with first and third, because it will be very hard to make people using our apps where people more trust in apps from big company

to solve that problem we can choose the second, there are many reason:
1. many of the apps not from big company, eventhough it's still skewed
2. it's look like people love religious apps, we can know from bible, so we can make apps which have connection with religious apps
3. we also know that dictionary book become dominance to in the second, which is make dictionary apps become potential

### Popular apps in google play

 to know which popular genre-apps in googleplay store, we can using install number

In [31]:
#using install number

display_table(googleplay_final, 5)

1,000,000+ :  15.7265
100,000+ :  11.5523
10,000,000+ :  10.5483
10,000+ :  10.1986
1,000+ :  8.3935
100+ :  6.9156
5,000,000+ :  6.8254
500,000+ :  5.5618
50,000+ :  4.7721
5,000+ :  4.5126
10+ :  3.5424
500+ :  3.2491
50,000,000+ :  2.3014
100,000,000+ :  2.1322
50+ :  1.9179
5+ :  0.7897
1+ :  0.5077
500,000,000+ :  0.2708
1,000,000,000+ :  0.2256
0+ :  0.0451
0 :  0.0113


In [32]:
#same as applestore apps
categories_googleplay = freq_table(googleplay_final, 1)
dict_avg = {}
for categories in categories_googleplay:
    total = 0
    len_category = 0
    for app in googleplay_final:
        category_app = app[1]
        
        #because n_instals in string type, we should change it to float
        #so we can calculate the average, to know which apps is popular
        if category_app == categories:
            n_instals = app[5]
            n_instals = n_instals.replace('+','')
            n_instals = n_instals.replace(',','')
            n_instals = float(n_instals)
            total += n_instals
            len_category +=1
    
    avg_instals = total / len_category
    dict_avg[categories] = avg_instals
    
    #make new list, this is have function to sorted the data from highest
    table_avg_tuple = []
    for key in dict_avg:
        key_val_as_tuple = (dict_avg[key], key)
        table_avg_tuple.append(key_val_as_tuple)
        
    #sorted table
    table_sorted = sorted(table_avg_tuple, reverse = True)
   
#print the data
for x in table_sorted:
    print(x[1], ':', x[0])

COMMUNICATION : 38456119.167247385
VIDEO_PLAYERS : 24727872.452830188
SOCIAL : 23253652.127118643
PHOTOGRAPHY : 17840110.40229885
PRODUCTIVITY : 16787331.344927534
GAME : 15588015.603248259
TRAVEL_AND_LOCAL : 13984077.710144928
ENTERTAINMENT : 11640705.88235294
TOOLS : 10801391.298666667
NEWS_AND_MAGAZINES : 9549178.467741935
BOOKS_AND_REFERENCE : 8767811.894736841
SHOPPING : 7036877.311557789
PERSONALIZATION : 5201482.6122448975
WEATHER : 5074486.197183099
HEALTH_AND_FITNESS : 4188821.9853479853
MAPS_AND_NAVIGATION : 4056941.7741935486
FAMILY : 3695641.8198090694
SPORTS : 3638640.1428571427
ART_AND_DESIGN : 1986335.0877192982
FOOD_AND_DRINK : 1924897.7363636363
EDUCATION : 1833495.145631068
BUSINESS : 1712290.1474201474
LIFESTYLE : 1437816.2687861272
FINANCE : 1387692.475609756
HOUSE_AND_HOME : 1331540.5616438356
DATING : 854028.8303030303
COMICS : 817657.2727272727
AUTO_AND_VEHICLES : 647317.8170731707
LIBRARIES_AND_DEMO : 638503.734939759
PARENTING : 542603.6206896552
BEAUTY : 51315

for the next analyze we focus on  coomunication with 38 million installs,

In [33]:
for app in googleplay_final:
    if app[1] == 'COMMUNICATION' and ( app[5] == '1,000,000,000+' or app[5] == '500,000,000+' or app[5] == '100,000,000+' or app[5] == '50,000,000+') :
        print(app[0], ':', app[5])

WhatsApp Messenger : 1,000,000,000+
imo beta free calls and text : 100,000,000+
Contacts : 50,000,000+
Android Messages : 100,000,000+
Google Duo - High Quality Video Calls : 500,000,000+
Messenger – Text and Video Chat for Free : 1,000,000,000+
imo free video calls and chat : 500,000,000+
free video calls and chat : 50,000,000+
Skype - free IM & video calls : 1,000,000,000+
Who : 100,000,000+
GO SMS Pro - Messenger, Free Themes, Emoji : 100,000,000+
LINE: Free Calls & Messages : 500,000,000+
Google Chrome: Fast & Secure : 1,000,000,000+
Firefox Browser fast & private : 100,000,000+
Dolphin Browser - Fast, Private & Adblock🐬 : 50,000,000+
UC Browser - Fast Download Private & Secure : 500,000,000+
Gmail : 1,000,000,000+
Mail.Ru - Email App : 50,000,000+
Hangouts : 1,000,000,000+
Azar : 50,000,000+
Messenger Lite: Free Calls & Messages : 100,000,000+
Kik : 100,000,000+
KakaoTalk: Free Calls & Text : 100,000,000+
Opera Mini - fast web browser : 100,000,000+
Opera Browser: Fast and Secure 

it's looks like same as social networking apps store, when apps with more 50 million install, dominate by apps from big company, such as whatsapps , google duo and many more.

let's we see another

In [34]:
for app in googleplay_final:
    if app[1] == 'SOCIAL' and ( app[5] == '1,000,000,000+' or app[5] == '500,000,000+' or app[5] == '100,000,000+' or app[5] == '50,000,000+') :
        print(app[0], ':', app[5])

Facebook : 1,000,000,000+
Facebook Lite : 500,000,000+
Tumblr : 100,000,000+
Pinterest : 100,000,000+
Google+ : 1,000,000,000+
Badoo - Free Chat & Dating App : 100,000,000+
Tango - Live Video Broadcast : 100,000,000+
ooVoo Video Calls, Messaging & Stories : 50,000,000+
Instagram : 1,000,000,000+
Snapchat : 500,000,000+
MeetMe: Chat & Meet New People : 50,000,000+
LinkedIn : 100,000,000+
Zello PTT Walkie Talkie : 50,000,000+
POF Free Dating App : 50,000,000+
SKOUT - Meet, Chat, Go Live : 50,000,000+
Tik Tok - including musical.ly : 100,000,000+
BIGO LIVE - Live Stream : 100,000,000+
VK : 100,000,000+


same as communication, dominate by apps from big company

In [35]:
for app in googleplay_final:
    if app[1] == 'VIDEO_PLAYERS' and ( app[5] == '1,000,000,000+' or app[5] == '500,000,000+' or app[5] == '100,000,000+' or app[5] == '50,000,000+') :
        print(app[0], ':', app[5])

YouTube : 1,000,000,000+
Motorola Gallery : 100,000,000+
VLC for Android : 100,000,000+
Vote for : 50,000,000+
Vigo Video : 50,000,000+
Google Play Movies & TV : 1,000,000,000+
MiniMovie - Free Video and Slideshow Editor : 50,000,000+
Samsung Video Library : 50,000,000+
LIKE – Magic Video Maker & Community : 50,000,000+
MX Player : 500,000,000+
Dubsmash : 100,000,000+
DU Recorder – Screen Recorder, Video Editor, Live : 50,000,000+
KineMaster – Pro Video Editor : 50,000,000+
VMate : 50,000,000+
HD Video Downloader : 2018 Best video mate : 50,000,000+
VivaVideo - Video Editor & Photo Movie : 100,000,000+
VideoShow-Video Editor, Video Maker, Beauty Camera : 100,000,000+
Ringdroid : 50,000,000+
Motorola FM Radio : 100,000,000+


In [36]:
for app in googleplay_final:
    if app[1] == 'PRODUCTIVITY' and ( app[5] == '1,000,000,000+' or app[5] == '500,000,000+' or app[5] == '100,000,000+' or app[5] == '50,000,000+') :
        print(app[0], ':', app[5])

Microsoft Word : 500,000,000+
Microsoft Outlook : 100,000,000+
Microsoft OneDrive : 100,000,000+
Calculator - unit converter : 50,000,000+
Microsoft OneNote : 100,000,000+
Google Keep : 100,000,000+
ES File Explorer File Manager : 100,000,000+
Dropbox : 500,000,000+
Google Docs : 100,000,000+
Advanced Task Killer : 50,000,000+
Microsoft PowerPoint : 100,000,000+
My Airtel-Online Recharge, Pay Bill, Wallet, UPI : 50,000,000+
Samsung Notes : 100,000,000+
Do It Later: Tasks & To-Dos : 50,000,000+
Verizon Cloud : 50,000,000+
myAT&T : 50,000,000+
SwiftKey Keyboard : 100,000,000+
MEGA : 50,000,000+
File Browser by Astro (File Manager) : 50,000,000+
Google Drive : 1,000,000,000+
Adobe Acrobat Reader : 100,000,000+
Google Sheets : 100,000,000+
Microsoft Excel : 100,000,000+
WPS Office - Word, Docs, PDF, Note, Slide & Sheet : 100,000,000+
Google Slides : 100,000,000+
ColorNote Notepad Notes : 100,000,000+
Evernote – Organizer, Planner for Notes & Memos : 100,000,000+
Google Calendar : 500,000,0

In [37]:
for app in googleplay_final:
    if app[1] == 'GAME' and ( app[5] == '1,000,000,000+' or app[5] == '500,000,000+' or app[5] == '100,000,000+' or app[5] == '50,000,000+') :
        print(app[0], ':', app[5])

Sonic Dash : 100,000,000+
PAC-MAN : 100,000,000+
Bubble Witch 3 Saga : 50,000,000+
Roll the Ball® - slide puzzle : 100,000,000+
Block Craft 3D: Building Simulator Games For Free : 50,000,000+
Love Balls : 50,000,000+
Piano Tiles 2™ : 100,000,000+
Pokémon GO : 100,000,000+
Snake VS Block : 50,000,000+
Extreme Car Driving Simulator : 100,000,000+
Trivia Crack : 100,000,000+
Angry Birds 2 : 100,000,000+
PUBG MOBILE : 50,000,000+
Summoners War : 50,000,000+
Lords Mobile: Battle of the Empires - Strategy RPG : 50,000,000+
Candy Crush Saga : 500,000,000+
8 Ball Pool : 100,000,000+
Subway Surfers : 1,000,000,000+
Candy Crush Soda Saga : 100,000,000+
Toy Blast : 50,000,000+
Clash Royale : 100,000,000+
Clash of Clans : 100,000,000+
Plants vs. Zombies FREE : 100,000,000+
Pou : 500,000,000+
Flow Free : 100,000,000+
My Talking Angela : 100,000,000+
slither.io : 100,000,000+
Cooking Fever : 100,000,000+
Yes day : 100,000,000+
Gardenscapes : 50,000,000+
Score! Hero : 100,000,000+
Magic Tiles 3 : 50,

 genre with average more 10 million installs , dominated by apps from big company , have same pattern with communication

from the last, let's we see reference, since that genre have potential in apple store

In [38]:
for app in googleplay_final:
    if app[1] == 'BOOKS_AND_REFERENCE' :
        print(app[0], ':', app[5])

E-Book Read - Read Book for free : 50,000+
Download free book with green book : 100,000+
Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Free Panda Radio Music : 100,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
English Grammar Complete Handbook : 500,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
Google Play Books : 1,000,000,000+
AlReader -any text book reader : 5,000,000+
Offline English Dictionary : 100,000+
Offline: English to Tagalog Dictionary : 500,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
Recipes of Prophetic Medicine for free : 500,000+
ReadEra – free ebook reader : 1,000,000+
Anonymous caller detection : 10,000+
Ebook Reader : 5,000,000+
Litnet - E-books : 100,000+
Read books online : 5,000,000+
English to Urdu Dictionary : 500,000+
eBoox: book reader fb2 epub zip : 1,000,000+
English Persian Dictionary : 500,000+
Flybook : 500,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
E

from more focus

In [39]:

for app in googleplay_final:
    if app[1] == 'BOOKS_AND_REFERENCE' and (app[5] == '1,000,000,000+'
                                            or app[5] == '500,000,000+'
                                            or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

Google Play Books : 1,000,000,000+
Bible : 100,000,000+
Amazon Kindle : 100,000,000+
Wattpad 📖 Free Books : 100,000,000+
Audiobooks from Audible : 100,000,000+


it's look like not too many apps from big company, let's we see another

In [40]:

for app in googleplay_final:
    if app[1] == 'BOOKS_AND_REFERENCE' and (app[5] == '1,000,000+'
                                            or app[5] == '5,000,000+'
                                            or app[5] == '10,000,000+'):
        print(app[0], ':', app[5])

Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
AlReader -any text book reader : 5,000,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
ReadEra – free ebook reader : 1,000,000+
Ebook Reader : 5,000,000+
Read books online : 5,000,000+
eBoox: book reader fb2 epub zip : 1,000,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
Moon+ Reader : 10,000,000+
English-Myanmar Dictionary : 1,000,000+
Golden Dictionary (EN-AR) : 1,000,000+
All Language Translator Free : 1,000,000+
Aldiko Book Reader : 10,000,000+
Dictionary - WordWeb : 5,000,000+
50000 Free eBooks & Free AudioBooks : 5,000,000+
Al-Quran (Free) : 10,000,000+
Al Quran Indonesia : 10,000,000+
Al'Quran Bahasa Indonesia : 10,000,000+
Al Quran Al karim : 1,000,000+
Al Quran : EAlim - Translations & MP3 Offline : 5,000,000+
Koran Read &MP3 30 Juz Offline : 1,000,000+
H

from this genre, we know that apps from big company not as much as genre with  average more than 10 million install. same as apple store, this genre have big potential,

We concluded that taking a popular book (perhaps a more recent book) and turning it into an app could be profitable for both the Google Play and the App Store markets. The markets are already full of libraries, so we need to add some special features besides the raw version of the book. This might include daily quotes from the book, an audio version of the book, quizzes on the book, a forum where people can discuss the book, etc.

## Most Favorite Genre

in this part i will try to find out most favorite genre and apps  using ratings

because many of ratings in googleplay is nan and we cannot calculate data with nan, i will replace nan with 0

In [41]:

import numpy as np

#make new_list to distict it with source data
no_nan =[]
for app in googleplay_final:
    no_nan.append(app)
    
#replace nan with 0
for app in no_nan:
    if app[2] == 'NaN':
        app[2] = '0'

display_table(no_nan,2)

0 :  14.6435
4.3 :  9.5217
4.4 :  9.2847
4.5 :  8.8109
4.2 :  8.4725
4.6 :  6.8366
4.1 :  6.6787
4.0 :  5.5167
4.7 :  4.3208
3.9 :  3.8696
3.8 :  3.0122
5.0 :  2.7414
3.7 :  2.403
4.8 :  2.0645
3.6 :  1.7712
3.5 :  1.6471
3.4 :  1.3087
3.3 :  1.0718
4.9 :  0.8912
3.0 :  0.8236
3.1 :  0.7333
3.2 :  0.6882
2.9 :  0.44
2.8 :  0.4174
2.6 :  0.2482
2.7 :  0.2369
2.5 :  0.2144
2.3 :  0.2031
2.4 :  0.1918
2.2 :  0.1579
1.0 :  0.1579
2.0 :  0.1241
1.9 :  0.1241
2.1 :  0.0903
1.8 :  0.079
1.7 :  0.079
1.6 :  0.0451
1.5 :  0.0338
1.4 :  0.0338
1.2 :  0.0113


In [42]:
#calculate average rating, depends on genre
categories_googleplay = freq_table(no_nan, 1)
dict_avg = {}

for categories in categories_googleplay:
    total = 0
    len_category = 0
    for app in googleplay_final:
        category_app = app[1]
        
        #because rating in string type, we should change it to float
        #so we can calculate the average, to know which apps is popular
        if category_app == categories:
            rating = app[2]
            rating = float(rating)
            total += rating
            len_category +=1
    
    avg_rating = total / len_category
    dict_avg[categories] = avg_rating
    
    #make new list, this is have function to sorted the data from highest
    table_avg_tuple = []
    for key in dict_avg:
        key_val_as_tuple = (dict_avg[key], key)
        table_avg_tuple.append(key_val_as_tuple)
        
    #sorted table
    table_sorted = sorted(table_avg_tuple, reverse = True)
   
#print the data
for x in table_sorted:
    print(x[1], ':', x[0])

EDUCATION : 4.298058252427182
ART_AND_DESIGN : 4.185964912280701
ENTERTAINMENT : 4.118823529411763
GAME : 4.030742459396756
COMICS : 4.025454545454546
PHOTOGRAPHY : 3.957088122605364
WEATHER : 3.871830985915492
SHOPPING : 3.781407035175881
FAMILY : 3.6934964200477376
VIDEO_PLAYERS : 3.6874213836477985
AUTO_AND_VEHICLES : 3.674390243902439
MAPS_AND_NAVIGATION : 3.648387096774193
BOOKS_AND_REFERENCE : 3.638421052631579
FINANCE : 3.6375000000000006
SOCIAL : 3.6220338983050833
HEALTH_AND_FITNESS : 3.615384615384615
PARENTING : 3.5913793103448284
TOOLS : 3.5284000000000004
TRAVEL_AND_LOCAL : 3.517874396135265
FOOD_AND_DRINK : 3.4854545454545454
HOUSE_AND_HOME : 3.4602739726027405
PRODUCTIVITY : 3.4182608695652217
PERSONALIZATION : 3.4078231292517014
BEAUTY : 3.3905660377358484
COMMUNICATION : 3.364808362369337
SPORTS : 3.3308970099667774
LIFESTYLE : 3.291618497109824
NEWS_AND_MAGAZINES : 3.277016129032258
LIBRARIES_AND_DEMO : 3.2216867469879515
EVENTS : 3.168253968253969
DATING : 3.16181818

i find that Education , art, and game have highest rating,
in this section i just focus on categories with highest average rating, so i will focus on education

In [43]:
for app in googleplay_final:
    if app[1] == 'EDUCATION':
        print(app[0], ':', app[12])

English Communication - Learn English for Chinese (Learn English for Chinese) : 4.0 and up
Khan Academy : 4.1 and up
Ai La Trieu Phu - ALTP Free : 4.0 and up
Learn Spanish - Español : 4.0 and up
Speed Reading : 4.0 and up
English for beginners : 4.0 and up
Mermaids : 4.1 and up
Learn Japanese, Korean, Chinese Offline & Free : 4.2 and up
Kids Mode : Varies with device
Dinosaurs Coloring Pages : 4.0 and up
Cars Coloring Pages : 4.0 and up
Math Tricks : 4.0 and up
Learn English Words Free : 4.1 and up
Japanese / English one-shop search dictionary - Free Japanese - English - Japanese dictionary application : 5.0 and up
English speaking texts : 1.6 and up
Thai Handwriting : 4.4 and up
THAI DICT 2018 : Varies with device
Kanji test · Han search Kanji training (free version) : 4.0.3 and up
Flippy Campus - Buy & sell on campus at a discount : 4.2 and up
Free intellectual training game application | : 4.4 and up
ABC Preschool Free : 2.3 and up
PINKFONG Baby Shark : 4.1 and up
English words appl

for education, apps like udemy, udacity whic are platform for online ciourse have high rating, followed by many dictionary and translate apps. 

if we want to make revenue via in-app purchases and subscriptions making paltform for online course have high potential.

in the most popular apps section, we know that book and references have high potential, so, although book and references have lower rating, let's we see if they have some interesting pattern

In [44]:
for app in googleplay_final:
    if app[1] == 'BOOKS_AND_REFERENCE':
        print(app[0], ':', app[12])

E-Book Read - Read Book for free : 4.4 and up
Download free book with green book : 4.0 and up
Wikipedia : Varies with device
Cool Reader : 1.5 and up
Free Panda Radio Music : 4.1 and up
Book store : 4.0 and up
FBReader: Favorite Book Reader : Varies with device
English Grammar Complete Handbook : 4.0 and up
Free Books - Spirit Fanfiction and Stories : 4.0.3 and up
Google Play Books : Varies with device
AlReader -any text book reader : 2.3 and up
Offline English Dictionary : 4.0.3 and up
Offline: English to Tagalog Dictionary : 4.0.3 and up
FamilySearch Tree : 4.1 and up
Cloud of Books : 4.1 and up
Recipes of Prophetic Medicine for free : 2.3 and up
ReadEra – free ebook reader : 4.1 and up
Anonymous caller detection : 2.3 and up
Ebook Reader : 4.0 and up
Litnet - E-books : 4.0 and up
Read books online : 4.1 and up
English to Urdu Dictionary : 4.0.3 and up
eBoox: book reader fb2 epub zip : 3.0 and up
English Persian Dictionary : 4.0.3 and up
Flybook : 2.1 and up
All Maths Formulas : 4.4 

same as before dictionary and plus many book about programming language have big potential to make revenue, it is because many of that have rating 4.0 or higher

In [45]:
#popular apps in applestore
categories_applestore = freq_table(applestore_final, -5)
dict_avg = {}

for categories in categories_applestore:
    total = 0
    len_category = 0
    for app in applestore_final:
        category_app = app[-5]
        
        #because n_instals in string type, we should change it to float
        #so we can calculate the average, to know which apps is popular
        if category_app == categories:
            rating = app[7]
            rating = float(rating)
            total += rating
            len_category +=1
    
    avg_rating = total / len_category
    dict_avg[categories] = avg_rating
    
    #make new list, this is have function to sorted the data from highest
    table_avg_tuple = []
    for key in dict_avg:
        key_val_as_tuple = (dict_avg[key], key)
        table_avg_tuple.append(key_val_as_tuple)
        
    #sorted table
    table_sorted = sorted(table_avg_tuple, reverse = True)
   
#print the data
for x in table_sorted:
    print(x[1], ':', x[0])

Catalogs : 4.125
Games : 4.037086446104589
Productivity : 4.0
Business : 3.9705882352941178
Shopping : 3.9702380952380953
Music : 3.946969696969697
Photo & Video : 3.903125
Navigation : 3.8333333333333335
Health & Fitness : 3.769230769230769
Reference : 3.6666666666666665
Education : 3.635593220338983
Food & Drink : 3.6346153846153846
Social Networking : 3.5943396226415096
Entertainment : 3.5393700787401574
Utilities : 3.5308641975308643
Travel : 3.4875
Weather : 3.482142857142857
Lifestyle : 3.411764705882353
Finance : 3.375
News : 3.244186046511628
Book : 3.0714285714285716
Sports : 3.0652173913043477
Medical : 3.0


in applestore , catalogs have highest average rating, let's see more deeply

In [46]:
applestore_data_header

['id',
 'track_name',
 'size_bytes',
 'currency',
 'price',
 'rating_count_tot',
 'rating_count_ver',
 'user_rating',
 'user_rating_ver',
 'ver',
 'cont_rating',
 'prime_genre',
 'sup_devices.num',
 'ipadSc_urls.num',
 'lang.num',
 'vpp_lic']

In [51]:
for app in applestore_final:
    if app[-5] == 'Catalogs':
        print(app[1], ':', app[7])

CPlus for Craigslist app - mobile classifieds : 5.0
DRAGONS MODS FREE for Minecraft PC Game Edition : 4.0
Face Swap and Copy Free – Switch & Fusion Faces in a Photo : 4.0
Ringtone Remixes - Marimba Remix Ringtones : 3.5


it's look in catalog genre, number of appas is not big enough to make conclution, so will try another genre

In [53]:
for app in applestore_final:
    if app[-5] == 'Games':
        print(app[1], ':', app[7], 'n_reviews : ', app[5])

Clash of Clans : 4.5 n_reviews :  2130805
Temple Run : 4.5 n_reviews :  1724546
Candy Crush Saga : 4.5 n_reviews :  961794
Angry Birds : 4.5 n_reviews :  824451
Subway Surfers : 4.5 n_reviews :  706110
Solitaire : 4.5 n_reviews :  679055
CSR Racing : 4.5 n_reviews :  677247
Crossy Road - Endless Arcade Hopper : 4.5 n_reviews :  669079
Injustice: Gods Among Us : 4.5 n_reviews :  612532
Hay Day : 4.5 n_reviews :  567344
PAC-MAN : 3.0 n_reviews :  508808
DragonVale : 4.5 n_reviews :  503230
Head Soccer : 5.0 n_reviews :  481564
Despicable Me: Minion Rush : 4.5 n_reviews :  464312
The Sims™ FreePlay : 4.5 n_reviews :  446880
Sonic Dash : 4.5 n_reviews :  418033
8 Ball Pool™ : 4.5 n_reviews :  416736
Tiny Tower - Free City Building : 4.5 n_reviews :  414803
Jetpack Joyride : 4.5 n_reviews :  405647
Bike Race - Top Motorcycle Racing Games : 4.5 n_reviews :  405007
Kim Kardashian: Hollywood : 4.5 n_reviews :  397730
Trivia Crack : 4.5 n_reviews :  393469
WordBrain : 4.5 n_reviews :  391401
Sn

it's look like high rating come from popular game which have high number of reviews,  for the last i will see productivity

In [54]:
for app in applestore_final:
    if app[-5] == 'Productivity':
        print(app[1], ':', app[7])

Evernote - stay organized : 4.0
Gmail - email by Google: secure, fast & organized : 4.0
iTranslate - Language Translator & Dictionary : 3.5
Yahoo Mail - Keeps You Organized! : 4.0
Google Docs : 4.5
Google Drive - free online storage : 4.5
Dropbox : 3.5
Microsoft Word : 4.5
Microsoft OneNote : 4.5
Microsoft Outlook - email and calendar : 3.5
Hotspot Shield Free VPN Proxy & Wi-Fi Privacy : 4.5
Documents 6 - File manager, PDF reader and browser : 4.5
Google Sheets : 4.5
Microsoft Excel : 4.5
Inbox by Gmail : 4.5
T-Mobile : 4.0
Paper by FiftyThree - Sketch, Diagram, Take Notes : 4.0
MyScript Calculator - Handwriting calculator : 4.5
VPN Proxy Master - Unlimited WiFi security VPN : 5.0
Microsoft OneDrive – File & photo cloud storage : 4.0
Ever - Capture Your Memories : 4.5
Speak & Translate － Voice and Text Translator : 4.5
Tayasui Sketches : 4.5
Drawing Desk - Draw, Paint, Doodle & Sketch board : 4.0
Microsoft PowerPoint : 4.0
Email - Fast & Secure mail for Gmail iCloud Yahoo : 4.5
Google 

from this point of view,  i think that if we want to make revenue, making games, or productivity apps or catalog apps have high potential

let's see from most popular data

In [55]:
for app in applestore_final:
    if app[-5] == 'Reference':
        print(app[1], ':', app[7])

Bible : 4.5
Dictionary.com Dictionary & Thesaurus : 4.0
Dictionary.com Dictionary & Thesaurus for iPad : 4.5
Google Translate : 3.5
Muslim Pro: Ramadan 2017 Prayer Times, Azan, Quran : 4.5
New Furniture Mods - Pocket Wiki & Game Tools for Minecraft PC Edition : 4.5
Merriam-Webster Dictionary : 4.5
Night Sky : 4.5
City Maps for Minecraft PE - The Best Maps for Minecraft Pocket Edition (MCPE) : 4.0
LUCKY BLOCK MOD ™ for Minecraft PC Edition - The Best Pocket Wiki & Mods Installer Tools : 4.0
GUNS MODS for Minecraft PC Edition - Mods Tools : 4.0
Guides for Pokémon GO - Pokemon GO News and Cheats : 4.5
WWDC : 3.5
Horror Maps for Minecraft PE - Download The Scariest Maps for Minecraft Pocket Edition (MCPE) Free : 4.0
VPN Express : 4.5
Real Bike Traffic Rider Virtual Reality Glasses : 3.0
教えて!goo : 0.0
Jishokun-Japanese English Dictionary & Translator : 0.0


if we see more deeply in references, although refrences have small rating, it's caused by many apps with have zero rating, but more important, same as most popular, apps like bible and dictionary have high potential to make revenue, because they have rating +- 4.5 